### ISSUE TITLE
TypeError: Cannot read property 'set' of undefined

### ISSUE THREAD

**Issue by deadcoder0904** [vvo/next-iron-session#404]
I am unable to set `session` as it is `undefined`.

I am using it with `next-connect` in my `handler` like:

```ts
import { NextApiResponse } from 'next'
import cookieSession from 'cookie-session'
import nc from 'next-connect'
import { ironSession } from 'next-iron-session'
import { error } from 'next/dist/build/output/log'
import { NextIronRequest } from '../types/index'

const COOKIE_SECRET = process.env.COOKIE_SECRET
const SESSION_SECRET = process.env.SESSION_SECRET

const IS_PRODUCTION = process.env.NODE_ENV !== 'development'

/**
 * Create an API route handler with next-connect and all the necessary middlewares
 *
 * @example
 * ```ts
 * export default handler().get((req, res) => { ... })
 * ```
 */
function handler() {
  if (!COOKIE_SECRET || !SESSION_SECRET)
    throw new Error(
      `Please add COOKIE_SECRET & SESSION_SECRET to your .env.local file!`
    )

  return nc<NextIronRequest, NextApiResponse>({
    onError: (err, _, res) => {
      error(err)
      res.status(500).end(err.toString())
    },
  })
    .use(
      cookieSession({
        name: 'session',
        keys: [COOKIE_SECRET],
        maxAge: 24 * 60 * 60 * 1000 * 30,
        secure: IS_PRODUCTION && !process.env.INSECURE_AUTH,
        signed: IS_PRODUCTION && !process.env.INSECURE_AUTH,
      })
    )
    .use(
      ironSession({
        cookieName: 'mysite-session',
        password: SESSION_SECRET,
        // if your localhost is served on http:// then disable the secure flag
        cookieOptions: {
          secure: IS_PRODUCTION,
        },
      })
    )
}

export default handler
```

Then I'm trying to use it in my api like:

```ts
import { NextApiResponse } from 'next'
import TwitterApi from 'twitter-api-v2'

import { TWITTER_CONFIG } from '../../../lib/config'
import { IRON_SESSION_ID_KEY, IS_PRODUCTION } from '../../../utils/index'
import { AppSession, NextIronRequest } from '../../../types/index'

export const CLIENT_URL = IS_PRODUCTION
  ? 'https://mysite.com/'
  : 'http://localhost:3000/'
export const SERVER_URL = IS_PRODUCTION
  ? 'https://mysite.com/api/'
  : 'http://localhost:3000/'

const generateAuthLink = async (
  req: NextIronRequest,
  res: NextApiResponse
) => {
  console.log('generateAuthLink')
  // Generate an authentication URL
  const { url, oauth_token, oauth_token_secret } = await new TwitterApi({
    appKey: TWITTER_CONFIG.consumerKey,
    appSecret: TWITTER_CONFIG.consumerSecret,
  }).generateAuthLink(SERVER_URL + '/api/twitter/get-verifier-token', {linkMode:'authorize'})

  console.log({url,oauth_token,oauth_token_secret})
  const reqWithSession = req as unknown as AppSession

  // store the relevant information in the session
  // session.twitterTokenSecret = req.session.twitterTokenSecret || {}
  // session.twitterTokenSecret[oauth_token] = oauth_token_secret
  reqWithSession.session.set(IRON_SESSION_ID_KEY, {
    [oauth_token]: oauth_token_secret
  })
  await reqWithSession.session.save()

  // redirect to the authentication URL
  res.redirect(url)
}

export default generateAuthLink
```

However, it gives me error on `session.set`. Am I doing this wrong?

I have made a complete reproduction here → https://github.com/deadcoder0904/twitter-api-v2-3-legged-login-using-next-connect

**Comment by deadcoder0904** [comment]
I used `withSession` as a wrapper like https://github.com/vvo/next-iron-session/blob/3f506f796cfef8c1a1f1199e044c2a0d0e653cdd/examples/next-typescript/pages/api/logout.ts & it worked.

However, I'm wondering why my code didn't work? This is the commit that didn't work → https://github.com/deadcoder0904/twitter-api-v2-3-legged-login-using-next-connect/tree/e14d85378bb0aadc3fb0d96a4823a014e92e73f5

I didn't have HOC `withIronSession` but I did specify `ironSession` as a middleware.
